`_read_planar_code()` incorrectly treats Plantri format as text rather than binary

### Steps To Reproduce

Download the file https://users.cecs.anu.edu.au/~bdm/data/5reg_20-32.pc and run the code:
```
file = open("5reg_20-32.pc")
list( graphs._read_planar_code(file) )
```


### Expected Behavior

It is expected that we obtain a list of 6 graphs contained in `5reg_20-32.pc` in Plantri format.

### Actual Behavior

We get an error:

> ValueError: 10 and 6 are not neighbors but 10 is in the list associated with 6

Deep down, the reason is that, since we opened the file in text mode, `_read_planar_code` starts screwing things up as soon as it encounters code `0x0A` (encoding vertex number 10 in Plantri). However, apparent attempt to change the mode via `open("5reg_20-32.pc", "rb")` raises another error in [line 1493](https://github.com/sagemath/sage/blob/ab24dac430ea4d21cb34d4a264b052590b09a95d/src/sage/graphs/graph_generators.py#L1493):

> AssertionError: Not a valid planar code header

### Additional Information

Line 1493 should perform comparison to a *binary* string, that is, `header == b'>>planar_code<<'`. It will also ensure that the file is opened in *binary* mode, so an error message could be made a bit more general - like:
```
assert header == b'>>planar_code<<', 'Not a valid planar code header or file is not opened in binary mode'
```

### Environment

- **OS**: Ubuntu 24.04.2 LTS
- **Sage Version**: 10.7.beta1


### Checklist

- [x] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [x] I have read the documentation and troubleshoot guide

--- a/src/sage/graphs/graph_generators.py
+++ b/src/sage/graphs/graph_generators.py
@@ -1479,7 +1479,7 @@ class GraphGenerators:
     def _read_planar_code(self, code_input, immutable=False):
         r"""
         Return a generator for the plane graphs in planar code format in
-        the file code_input (see [BM2016]_).
+        the binary file ``code_input`` (see [BM2016]_).
 
         A file with planar code starts with a header ``>>planar_code<<``.
         After the header each graph is stored in the following way :
@@ -1493,7 +1493,7 @@ class GraphGenerators:
 
         INPUT:
 
-        - ``code_input`` -- a file containing valid planar code data
+        - ``code_input`` -- a binary file containing valid planar code data
 
         - ``immutable`` -- boolean (default: ``False``); whether to return
           immutable or mutable graphs
@@ -1513,18 +1513,17 @@ class GraphGenerators:
 
         EXAMPLES:
 
-        The following example creates a small planar code file in memory and
-        reads it using the ``_read_planar_code`` method::
+        The following example creates a small planar code binary
+        file in memory and reads it using the ``_read_planar_code`` method::
 
-            sage: from io import StringIO
-            sage: code_input = StringIO('>>planar_code<<')
-            sage: _ = code_input.write('>>planar_code<<')
+            sage: from io import BytesIO
+            sage: code_input = BytesIO()
+            sage: n = code_input.write(b'>>planar_code<<')
             sage: for c in [4,2,3,4,0,1,4,3,0,1,2,4,0,1,3,2,0]:
-            ....:     _ = code_input.write('{:c}'.format(c))
-            sage: _ = code_input.seek(0)
+            ....:     n = code_input.write(bytes('{:c}'.format(c),'ascii'))
+            sage: n = code_input.seek(0)
             sage: gen = graphs._read_planar_code(code_input)
-            sage: l = list(gen)
-            sage: l
+            sage: l = list(gen); l
             [Graph on 4 vertices]
             sage: l[0].is_isomorphic(graphs.CompleteGraph(4))
             True
@@ -1533,10 +1532,31 @@ class GraphGenerators:
              2: [1, 4, 3],
              3: [1, 2, 4],
              4: [1, 3, 2]}
+
+        TESTS::
+
+            sage: from io import StringIO
+            sage: code_input = StringIO()
+            sage: n = code_input.write('>>planar_code<<')
+            sage: list(graphs._read_planar_code(code_input))
+            Traceback (most recent call last):
+            ...
+            TypeError: not a binary file
+
+            sage: from io import BytesIO
+            sage: code_input = BytesIO()
+            sage: n = code_input.write(b'>>wrong header<<')
+            sage: list(graphs._read_planar_code(code_input))
+            Traceback (most recent call last):
+            ...
+            TypeError: file has no valid planar code header
         """
         # start of code to read planar code
         header = code_input.read(15)
-        assert header == '>>planar_code<<', 'Not a valid planar code header'
+        if not isinstance(header, bytes):
+            raise TypeError('not a binary file')
+        if header != b'>>planar_code<<':
+            raise TypeError('file has no valid planar code header')
 
         # read graph per graph
         while True:

--- a/src/sage/graphs/graph_generators.py
+++ b/src/sage/graphs/graph_generators.py
@@ -1538,6 +1538,7 @@ class GraphGenerators:
             sage: from io import StringIO
             sage: code_input = StringIO()
             sage: n = code_input.write('>>planar_code<<')
+            sage: n = code_input.seek(0)
             sage: list(graphs._read_planar_code(code_input))
             Traceback (most recent call last):
             ...
@@ -1546,6 +1547,7 @@ class GraphGenerators:
             sage: from io import BytesIO
             sage: code_input = BytesIO()
             sage: n = code_input.write(b'>>wrong header<<')
+            sage: n = code_input.seek(0)
             sage: list(graphs._read_planar_code(code_input))
             Traceback (most recent call last):
             ...
@@ -1709,10 +1711,7 @@ class GraphGenerators:
 
         sp = subprocess.Popen(command, shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE, close_fds=True,
-                              encoding='latin-1')
-
-        sp.stdout.reconfigure(newline='')
+                              stderr=subprocess.PIPE, close_fds=True)
 
         yield from graphs._read_planar_code(sp.stdout, immutable=immutable)
 
@@ -1800,10 +1799,7 @@ class GraphGenerators:
 
         sp = subprocess.Popen(command, shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE, close_fds=True,
-                              encoding='latin-1')
-
-        sp.stdout.reconfigure(newline='')
+                              stderr=subprocess.PIPE, close_fds=True)
 
         yield from graphs._read_planar_code(sp.stdout, immutable=immutable)
 
@@ -1988,10 +1984,7 @@ class GraphGenerators:
                                  options)
         sp = subprocess.Popen(command, shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE, close_fds=True,
-                              encoding='latin-1')
-
-        sp.stdout.reconfigure(newline='')
+                              stderr=subprocess.PIPE, close_fds=True)
 
         try:
             yield from graphs._read_planar_code(sp.stdout, immutable=immutable)

--- a/src/sage/graphs/graph_generators.py
+++ b/src/sage/graphs/graph_generators.py
@@ -35,11 +35,11 @@ def __append_to_doc(methods):
                 "    :widths: 33, 33, 33\n"
                 "    :delim: |\n\n")
 
-    h = (len(methods)+2)//3
+    h = (len(methods) + 2) // 3
     # Reorders the list of methods for horizontal reading, the only one Sphinx understands
-    reordered_methods = [0]*3*h
+    reordered_methods = [0] * (3 * h)
     for i, m in enumerate(methods):
-        reordered_methods[3*(i % h) + (i//h)] = m
+        reordered_methods[3 * (i % h) + (i // h)] = m
     methods = reordered_methods
 
     # Adding the list to the __doc__ string
@@ -827,7 +827,7 @@ class GraphGenerators:
             if vertices is None:
                 raise NotImplementedError
             if (len(degree_sequence) != vertices or sum(degree_sequence) % 2
-                    or sum(degree_sequence) > vertices*(vertices - 1)):
+                    or sum(degree_sequence) > vertices * (vertices - 1)):
                 raise ValueError("Invalid degree sequence.")
             degree_sequence = sorted(degree_sequence)
             if augment == 'edges':
@@ -1988,7 +1988,7 @@ class GraphGenerators:
 
         try:
             yield from graphs._read_planar_code(sp.stdout, immutable=immutable)
-        except AssertionError:
+        except (TypeError, AssertionError):
             raise AttributeError("invalid options '{}'".format(options))
 
     def planar_graphs(self, order, minimum_degree=None,
@@ -2191,7 +2191,7 @@ class GraphGenerators:
                     raise ValueError("the number of edges cannot be less than order - 1")
                 edges = '-e:{}'.format(maximum_edges)
         else:
-            if minimum_edges > 3*order - 6:
+            if minimum_edges > 3 * order - 6:
                 raise ValueError("the number of edges cannot be more than 3*order - 6")
             if maximum_edges is None:
                 edges = '-e{}:'.format(minimum_edges)
@@ -2952,7 +2952,7 @@ def canaug_traverse_vert(g, aut_gens, max_verts, property, dig=False, loops=Fals
         # in the case of graphs, there are n possibilities,
         # and in the case of digraphs, there are 2*n.
         if dig:
-            possibilities = 2*n
+            possibilities = 2 * n
         else:
             possibilities = n
         num_roots = 2**possibilities

